manifests/client.pp

```diff
@@ -44,6 +44,11 @@
     before => File['krb5.conf'],
   }
 
+  $krb5_conf_dir = dirname($krb5_conf_path)
+  if !defined(File[$krb5_conf_dir]) {
```

Comment:
The if statement is redundant. file{} will create the folder if it doesn't exist. Please simplify it.